Stop patching after bakery is closed

Suppose we have Cake with other cakes as parameters
```python
class MyBakery(Bakery):
    nested_cake: Any = Cake(Cake(None))
```
If we run tests with patching, they would fail
```python
async def test_my_bakery(bakery_mock):
    bakery_mock.nested_cake = Cake(None)
    async with bakery_mock(MyBakery):
        pass
```
with error
```shell
cake = Cake '<anon>'

    def assert_baked(cake: Cakeable[Any]) -> None:
        """Check if recipe is baked."""
        if not is_baked(cake):
>           raise ValueError(f"{cake} is not baked. Just bake it!")
E           ValueError: Cake '<anon>' is not baked. Just bake it!
```
Possible mocker.patch [should be reset](https://github.com/Mityuha/fresh-bakery/blob/main/bakery/testbakery.py#L131) after bakery closed.

--- a/bakery/testbakery.py
+++ b/bakery/testbakery.py
@@ -128,11 +128,11 @@ class BakeryMock:
 
     async def reset(self) -> None:
         """Stop patching."""
-        self._mocker_.stopall()
-
         if self._bakery_:
             await self._bakery_.aclose()
             self._bakery_ = None
+
+        self._mocker_.stopall()
 
     def __setattr__(self, attr: str, value: Any) -> None:
         if attr in (

--- a/tests/test_bakery_mock.py
+++ b/tests/test_bakery_mock.py
@@ -237,6 +237,10 @@ async def test_bakery_mock_replace_recipe(bakery_mock: BakeryMock) -> None:
 async def test_nested_cakes_mock_simple(bakery_mock: BakeryMock) -> None:
     """Test simple nested cakes mock."""
 
+    def make_list(*args: Any) -> List:
+        """Make list."""
+        return list(*args)
+
     class MyBakery(Bakery):
         """MyBakery."""
 
@@ -244,7 +248,7 @@ async def test_nested_cakes_mock_simple(bakery_mock: BakeryMock) -> None:
             sum,
             Cake(
                 Cake(
-                    list,
+                    make_list,
                     [Cake(sum, [1, 2, 3])],  # type: ignore
                 ),
             ),

--- a/tests/test_bakery_mock.py
+++ b/tests/test_bakery_mock.py
@@ -234,6 +234,34 @@ async def test_bakery_mock_replace_recipe(bakery_mock: BakeryMock) -> None:
         assert Comp2().comp.total == 600
 
 
+async def test_nested_cakes_mock_simple(bakery_mock: BakeryMock) -> None:
+    """Test simple nested cakes mock."""
+
+    class MyBakery(Bakery):
+        """MyBakery."""
+
+        value: float = Cake(
+            sum,
+            Cake(
+                Cake(
+                    list,
+                    [Cake(sum, [1, 2, 3])],  # type: ignore
+                ),
+            ),
+        )
+
+    async with MyBakery() as bakery:
+        assert bakery.value == 6
+
+    bakery_mock.value = Cake(10)
+    async with bakery_mock(MyBakery):
+        assert MyBakery().value == 10
+
+    bakery_mock.value = lambda *args: sum(*args, 4)  # type: ignore
+    async with bakery_mock(MyBakery):
+        assert MyBakery().value == 10
+
+
 async def test_nested_cakes_mock(bakery_mock: BakeryMock) -> None:
     """Test nested cakes mock."""
 

--- a/tests/test_bakery_mock.py
+++ b/tests/test_bakery_mock.py
@@ -3,7 +3,7 @@
 
 from dataclasses import dataclass
 from time import sleep
-from typing import Dict, List
+from typing import Any, Dict, List
 
 from bakery import Bakery, Cake, is_baked
 from bakery.testbakery import BakeryMock
@@ -232,3 +232,34 @@ async def test_bakery_mock_replace_recipe(bakery_mock: BakeryMock) -> None:
         assert Comp2().comp.sleeper2 == 200
         assert Comp2().comp.some_list[2] == 300
         assert Comp2().comp.total == 600
+
+
+async def test_nested_cakes_mock(bakery_mock: BakeryMock) -> None:
+    """Test nested cakes mock."""
+
+    class Database:
+        """Database."""
+
+        def __init__(self, table: str):
+            self.connected: bool = False
+            self.table: str = table
+
+        async def __aenter__(self) -> "Database":
+            self.connected = True
+            return self
+
+        async def __aexit__(self, *_args: Any) -> None:
+            self.connected = False
+
+    class MyBakery(Bakery):
+        """MyBakery."""
+
+        database: Database = Cake(Cake(Database, table=Cake("table")))
+
+    async with MyBakery() as bakery:
+        assert bakery.database.connected
+
+    bakery_mock.database = Cake(None)
+
+    async with bakery_mock(MyBakery):
+        assert MyBakery().database is None

